source/jenkins/upgrade-java.html.md.erb

```diff
@@ -0,0 +1,76 @@
+---
+title: Upgrade Java for Jenkins
+last_reviewed_on: 2024-07-15
+review_in: 12 months
+weight: 10
+---
+
+# <%= current_page.data.title %>
+
+It is always a good idea to test things in SBOX before doing them in PROD, upgrading Java for Jenkins is no exception.
+Upgrade the agents first and then the built in controller node.
+
+## Upgrade Jenkins agents
+
+1. Create a PR in the [CNP Jenkins repo](https://github.com/hmcts/cnp-jenkins-docker/blob/master/jenkins/Dockerfile)  with the Java version incremented. You can find updated Jenkins Base image tag from the [Jenkins dockerhub repo](https://hub.docker.com/r/jenkins/jenkins/tags).
+2. Creating the PR should have automatically created a docker image with a pr- in it's name.
+3. Check the [container registry](https://portal.azure.com/#@HMCTS.NET/resource/subscriptions/8999dec3-0104-4a27-94ee-6588559729d1/resourceGroups/rpe-acr-prod-rg/providers/Microsoft.ContainerRegistry/registries/hmctspublic/repository) for the updated tag.
+4. Alter the path to the java runtime to point to the upgraded java version for [CFT](https://github.com/hmcts/cnp-flux-config/blob/master/apps/jenkins/jenkins/sbox-intsvc/jenkins-azure-vm-agent.yaml) or [SDS](https://github.com/hmcts/sds-flux-config/blob/master/apps/jenkins/jenkins/sbox/jenkins-azure-vm-agent.yaml) if depending on what you are upgrading.
+5. After 10 - 15 minutes verify that the upgrade was successful by running [Plum tests](https://build.hmcts.net/job/HMCTS_a_to_c/job/cnp-plum-frontend/job/master/) or [Toffee tests](https://sds-build.hmcts.net/blue/organizations/jenkins/HMCTS_Nightly_Sandbox/sds-toffee-frontend/) if you are upgrading SDS.
+
+#### Production
+
+Same steps as above but instead use the following for [CFT](https://github.com/hmcts/cnp-flux-config/blob/master/apps/jenkins/jenkins/ptl-intsvc/jenkins-azure-vm-agent.yaml) or [SDS](https://github.com/hmcts/sds-flux-config/blob/master/apps/jenkins/jenkins/ptl/jenkins-azure-vm-agent.yaml)
```

Comment:
I think this production step should come at the end with the controller production step too -- i.e

Order:
- agent test in sbox with PR-* image
- controller test in sbox to test controller&agent compatability
Then
- if all happy, merge agent pr so we can have a new agent image
- upgrade agent image in flux for jenkins in prod env
- upgrade controller in jenkins prod env
- rollback etc

What do you think? Otherwise we may end up with production agent changes merged and added before we've even touched controller

Reply:
This stuff also looks like it's covered in the next steps too, like ` If you have not already done so create a PR in the [CNP Jenkins repo](https://github.com/hmcts/cnp-jenkins-docker/blob/master/jenkins/Dockerfile) with the Java version incremented.` and ` If you have not already done so create a PR in the [CNP Jenkins repo](https://github.com/hmcts/cnp-jenkins-docker/blob/master/jenkins/Dockerfile) with the Java version incremented.`

I think it could just be written in one nicer flow